### PR TITLE
Rust: add mentoring notes for atbash-cipher

### PR DESCRIPTION
**Pull request by yaahallo**

### CODE REVIEW

File: tracks/rust/exercises/atbash-cipher/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+- Use char methods for working with ascii data, all of which contain `_ascii_`
+  in their name, rather than filtering characters with `is_ascii` and using
+  unicode centric methods for manipulation.
+- Factor the common logic into a seperate function rather than duplicating
+  filtering and transposition in encode and decode. **Bonus** its fine if this
+  function returns a String or if they just use decode as the starting point of
+  encode, but this is good place to introduce the `impl Trait` in the return
+  position after having already approved the exercise.
+- **Optional**: Avoid heap allocations when inserting spaces into the encoded
+  final string. Its better to `flat_map` spaces into the than collecting each
```
Comment (NobbZ) [inline tracks/rust/exercises/atbash-cipher/mentoring.md]: “into the than” reads a bit weird for me. Can you rephrase that a bit, making sure it reads like a proper sentence for non-native speakers as well?

File: tracks/rust/exercises/atbash-cipher/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+use std::iter::once;
+
+static ALPHABET: &'static str = "abcdefghijklmnopqrstuvwxyz";
+
+/// "Encipher" with the Atbash cipher.
+pub fn encode(plain: &str) -> String {
+    atbash(plain)
+        .enumerate()
+        .flat_map(|(ind, c)| {
+            match ind % 5 {
```
Comment (NobbZ) [inline tracks/rust/exercises/atbash-cipher/mentoring.md]: I do not have the exact algorithm in my mind, but `5` feels quite like a magic number.

Comment (yaahallo) [reply tracks/rust/exercises/atbash-cipher/mentoring.md]: I believe 5 is the group size you're supposed to split the output into.

File: tracks/rust/exercises/atbash-cipher/mentoring.md (hunk 1 above)
Comment (yaahallo) [reply tracks/rust/exercises/atbash-cipher/mentoring.md]: It looks like its missing a word or two.

File: tracks/rust/exercises/atbash-cipher/mentoring.md (hunk 2 above)
Comment (NobbZ) [reply tracks/rust/exercises/atbash-cipher/mentoring.md]: Usually I'd ask students to use a constant in such cases.

Comment (yaahallo) [reply tracks/rust/exercises/atbash-cipher/mentoring.md]: I'll add it as a possible comment. But I would tend to see that as a style issue rather than a rust issue and I tend to be hesitant about giving those kinds of comments for fear of push back.
